framework/wcf/Extensibility/Transport/AzureQueueStorageClient/AzureQueueStorageClient/AzureQueueStorageTransportBindingElement.cs

```diff
@@ -0,0 +1,303 @@
+        {
+            return (typeof(TChannel) == typeof(IOutputChannel));
+        }
+
+        /// <summary>
+        /// Used by higher layers to determine what types of channel listeners this
+        /// binding element supports. Which in this case is just IInputChannel.
+        /// </summary>
+        /*public override bool CanBuildChannelListener<TChannel>(BindingContext context)
+        {
+            return (typeof(TChannel) == typeof(IInputChannel));
+        }*/
+
+        public override string Scheme
+        {
+            get
+            {
+                return AzureQueueStorageConstants.Scheme;
+            }
+        }
+
+        // We expose in policy the fact that we're UDP, and whether we're multicast or not.
+        // Import is done through UdpBindingElementImporter.
+        /*void IPolicyExportExtension.ExportPolicy(MetadataExporter exporter, PolicyConversionContext context)
+        {
+            if (exporter == null)
+            {
+                throw new ArgumentNullException("exporter");
+            }
+
+            if (context == null)
+            {
+                throw new ArgumentNullException("context");
+            }
+
+            ICollection<XmlElement> bindingAssertions = context.GetBindingAssertions();
+            XmlDocument xmlDocument = new XmlDocument();
+            bindingAssertions.Add(xmlDocument.CreateElement(
+                UdpPolicyStrings.Prefix, UdpPolicyStrings.TransportAssertion, UdpPolicyStrings.UdpNamespace));
+
+            if (Multicast)
+            {
+                bindingAssertions.Add(xmlDocument.CreateElement(
+                    UdpPolicyStrings.Prefix, UdpPolicyStrings.MulticastAssertion, UdpPolicyStrings.UdpNamespace));
+            }
+
+            bool createdNew = false;
+            MessageEncodingBindingElement encodingBindingElement = context.BindingElements.Find<MessageEncodingBindingElement>();
+            if (encodingBindingElement == null)
+            {
+                createdNew = true;
+                encodingBindingElement = new TextMessageEncodingBindingElement();
+            }
+
+            if (createdNew && encodingBindingElement is IPolicyExportExtension)
+            {
+                ((IPolicyExportExtension)encodingBindingElement).ExportPolicy(exporter, context);
+            }
+
+            AddWSAddressingAssertion(context, encodingBindingElement.MessageVersion.Addressing);
+        }*/
+
+        public override BindingElement Clone()
+        {
+            return new AzureQueueStorageTransportBindingElement(this);
+        }
+
+        public override T GetProperty<T>(BindingContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException("context");
+            }
+
+            return context.GetInnerProperty<T>();
+        }
+
+        /*public void ExportContract(WsdlExporter exporter, WsdlContractConversionContext context)
+        {
```

Comment:
You can remove all this commented out code from this line and below as well as the XmlDoc property at the bottom (and the field for it at the top).

Reply:
removed them.